### DISCUSSION TITLE
Vitepress is SSG，why I still need to make my codes works in SSR? but `import.meta.env.SSR` is false?

### DISCUSSION THREAD

**Discussion by ishowman** [vuejs/vitepress#3097] [Q&A]
- Vitepress is SSG，why I still need to make my codes works in SSR? I need to use `defineClientComponent` to solve API compatibility for build.
- Since I need to solve API compatibility for SSR, why `import.meta.env.SSR` is false? 

This is a conflic design.And `defineClientComponent` cause some problems such as I can't use variable via component's ref that defined with `defineExpose`.

**Comment by brc-dd** [comment]
> Vitepress is SSG，why I still need to make my codes works in SSR

Because to generate the static code we need to render your code on server first during build.

> why import.meta.env.SSR is false

It isn't during SSR phase. You're probably checking the value on dev or on client-side.

**Reply by ishowman** [reply]
@brc-dd   Thanks for your reply. Is it possible to  generate the static code without server render?（it will be great if it is decide by configuration）

**Reply by brc-dd** [reply]
It currently out of scope. You can directly use vite+vue starter with unplugin-vue-markdown if you need client-side rendering.
